selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/WorkspaceDetailsTest.java

```diff
@@ -0,0 +1,263 @@
+
+  @BeforeClass
+  public void setUp() throws Exception {
+    dashboard.open();
+    createWsFromJavaMySqlStack();
+  }
+
+  @AfterClass
+  public void tearDown() throws Exception {
+    workspaceServiceClient.delete(WORKSPACE, defaultTestUser.getName());
+  }
+
+  @Test(priority = 0)
+  public void workingWithEnvVariables() {
+    dashboardWorkspace.selectTabInWorspaceMenu(TabNames.ENV_VARIABLES);
+    dashboardWorkspace.selectMachine("Environment variables", "dev-machine");
+    dashboardWorkspace.clickOnAddEnvVariableButton();
+    dashboardWorkspace.checkAddNewEnvVarialbleDialogIsOpen();
+    dashboardWorkspace.addNewEnvironmentVariable("logi", "qaadmin");
+    dashboardWorkspace.clickOnAddDialogButton();
+    dashboardWorkspace.checkEnvVariableExists("logi");
+    dashboardWorkspace.checkValue("logi", "qaadmin");
+    dashboardWorkspace.clickOnEditEnvVariableButton("logi");
+    dashboardWorkspace.enterEnvVariableName("login");
+    dashboardWorkspace.clickOnUpdateDialogButton();
+    dashboardWorkspace.checkValue("login", "qaadmin");
+    dashboardWorkspace.clickOnEnvVariableCheckbox("login");
+    dashboardWorkspace.clickOnDeleteBtn();
+    dashboardWorkspace.clickOnDeleteDialogButton();
+    clickOnSaveButton();
+
+    dashboardWorkspace.selectMachine("Environment variables", "db");
+    for (String name : VariablesList) {
+      loader.waitOnClosed();
+      dashboardWorkspace.clickOnDeleteEnvVariableButton(name);
+      dashboardWorkspace.clickOnDeleteDialogButton();
+    }
+    clickOnSaveButton();
+
+    for (int i = 0; i < VariablesList.size(); i++) {
+      loader.waitOnClosed();
+      dashboardWorkspace.clickOnAddEnvVariableButton();
+      dashboardWorkspace.checkAddNewEnvVarialbleDialogIsOpen();
+      dashboardWorkspace.addNewEnvironmentVariable(VariablesList.get(i), ValuesList.get(i));
+      dashboardWorkspace.clickOnAddDialogButton();
+      dashboardWorkspace.checkEnvVariableExists(VariablesList.get(i));
+      dashboardWorkspace.checkValue(VariablesList.get(i), ValuesList.get(i));
+    }
+    clickOnSaveButton();
+  }
+
+  @Test(priority = 1)
+  public void workingWithAgents() {
+    Boolean b;
+    dashboardWorkspace.selectTabInWorspaceMenu(TabNames.AGENTS);
+    dashboardWorkspace.selectMachine("Workspace Agents", "dev-machine");
+    for (String agentName : AgentsList) {
+      dashboardWorkspace.checkAgentExists(agentName);
+    }
+
+    for (int i = 0; i < AgentsList.size(); i++) {
+      b = dashboardWorkspace.getAgentState(AgentsList.get(i));
+      Assert.assertEquals(b, AgentsStateList.get(i));
+      dashboardWorkspace.switchAgentState(AgentsList.get(i));
+    }
+    clickOnSaveButton();
+
+    for (String agentName : AgentsList) {
+      dashboardWorkspace.switchAgentState(agentName);
+    }
+    for (int i = 0; i < AgentsList.size(); i++) {
+      b = dashboardWorkspace.getAgentState(AgentsList.get(i));
+      Assert.assertEquals(b, AgentsStateList.get(i));
+    }
+    clickOnSaveButton();
```

Comment:
Why do we need to click on the save button twice? What is the test case above the code?

Reply:
In this test case is checking that agents switched to the previous states. The save button clicks first after switching agents and second time it clicks after agents were switch to the previous states(state by default after a workspace creation).

Reply:
Hm. it's not obvious from the test name *workingWithAgents*, and from the test content, as for me.
Let's rename the test, or comment on second part of test that it verifies switch to the previous states (default state after the workspace creation).